x/wasm/types/proposal.go

```diff
@@ -62,13 +62,13 @@ func init() { // register new content types with the sdk
 	govtypes.RegisterProposalType(string(ProposalTypeClearAdmin))
 	govtypes.RegisterProposalType(string(ProposalTypePinCodes))
 	govtypes.RegisterProposalType(string(ProposalTypeUnpinCodes))
-	govtypes.RegisterProposalTypeCodec(StoreCodeProposal{}, "wasm/StoreCodeProposal")
-	govtypes.RegisterProposalTypeCodec(InstantiateContractProposal{}, "wasm/InstantiateContractProposal")
-	govtypes.RegisterProposalTypeCodec(MigrateContractProposal{}, "wasm/MigrateContractProposal")
-	govtypes.RegisterProposalTypeCodec(UpdateAdminProposal{}, "wasm/UpdateAdminProposal")
-	govtypes.RegisterProposalTypeCodec(ClearAdminProposal{}, "wasm/ClearAdminProposal")
-	govtypes.RegisterProposalTypeCodec(PinCodesProposal{}, "wasm/PinCodesProposal")
-	govtypes.RegisterProposalTypeCodec(UnpinCodesProposal{}, "wasm/UnpinCodesProposal")
+	govtypes.RegisterProposalTypeCodec(&StoreCodeProposal{}, "wasm/StoreCodeProposal")
```

Comment:
I wondered about this (pointer or struct) issue before when you created it.
My suggestion is to register both, as you cannot be fully sure what will come down the pipe and I have been hit by panics when only one type was registered before (*Msg but not Msg). Maybe that was only with amino, but in general, seems safer to just keep the old registration and add these to it (or is it illegal to have 2 types with the same type name?)

Reply:
You can not register both but it is best practice in the sdk to register the pointers not structs anymore. For example:
* [params](https://github.com/cosmos/cosmos-sdk/blob/v0.42.3/x/params/types/proposal/codec.go#L11)
* [upgrade](https://github.com/cosmos/cosmos-sdk/blob/v0.42.3/x/upgrade/types/codec.go#L12-L13)